### DISCUSSION TITLE
AttributeError in fastapi: How to use one class attribute of a class as a type inside another class in fastapi?

### DISCUSSION THREAD

**Discussion by p0l4r** [fastapi/fastapi#3929] [Questions] [answered]
I am a newbie in fastapi. I am getting AttributeError while trying to assign one class attribute of a class as a type inside another class.  I have two different pydantic BaseModel class in two different files. 
**in X.py I have:**
```
from pydantic import BaseModel
from typing import Optional

class A(BaseModel):
    a: int
    b: Optional[str]
    c: bool = False
```

**in Y.py I have:**

```
from pydantic import BaseModel
from typing import Optional

class B(BaseModel):
    a: str
    b: Optional[str]
    
```

**what I want to do in Y.py is:**

```
class B(BaseModel):
    a: str
    b: Optional[str]
    c: A.c #I want to add the attribute "c" from class A as a type here 
```

**_I am getting AttributeError saying type obj "A" has no attribute "c"_**

Can anyone please help me with this?

**Comment by ghandic** [comment, marked as answer]
You could use inheritance? Obviously, that comes with its own quirks. What is the general use that you need this for? Typing out a few extra variables cant be to bad :) if you need to introduce tighter coupling then inheritance can do that for you

```python
from pydantic import BaseModel
from typing import Optional

class A(BaseModel):
    a: int
    b: Optional[str]
    c: bool = False


class B(A):
    a: str  # Note we are overriding "a" here
```

Which effectively will give you a class `B` with:

```python

class B:
    a: str
    b: Optional[str]
    c: bool = False
```

**Reply by ghandic** [reply]
@p0l4r  does this answer your question?

**Reply by p0l4r** [reply]
yes
